packages/web/components/drawers/token-select-drawer-limit.tsx

```diff
@@ -0,0 +1,445 @@
+import { Transition } from "@headlessui/react";
+import { CoinPretty, PricePretty } from "@keplr-wallet/unit";
+import classNames from "classnames";
+import { observer } from "mobx-react-lite";
+import Image from "next/image";
+import { Fragment, FunctionComponent, useRef, useState } from "react";
+import { useLatest } from "react-use";
+
+import { Icon } from "~/components/assets";
+import IconButton from "~/components/buttons/icon-button";
+import { SearchBox } from "~/components/input";
+import { Tooltip } from "~/components/tooltip";
+import { useTranslation } from "~/hooks";
+import { useWindowSize } from "~/hooks";
+import { useSwapAsset } from "~/hooks/use-swap";
+import { ActivateUnverifiedTokenConfirmation } from "~/modals";
+import { UnverifiedAssetsState } from "~/stores/user-settings";
+import { formatPretty } from "~/utils/formatter";
+
+import { useConst } from "../../hooks/use-const";
+import useDraggableScroll from "../../hooks/use-draggable-scroll";
+import { useKeyActions } from "../../hooks/use-key-actions";
+import { useStateRef } from "../../hooks/use-state-ref";
+import { useWindowKeyActions } from "../../hooks/window/use-window-key-actions";
+import { useStore } from "../../stores";
+import { Intersection } from "../intersection";
+import Spinner from "../loaders/spinner";
+
+const dataAttributeName = "data-token-id";
+
+function getTokenItemId(uniqueId: string, index: number) {
+  return `token-selector-item-${uniqueId}-${index}`;
+}
+
+function getTokenElement(uniqueId: string, index: number) {
+  return document.querySelector(
+    `[${dataAttributeName}=${getTokenItemId(uniqueId, index)}]`
+  );
+}
+
+function getAllTokenElements() {
+  return document.querySelectorAll(`[${dataAttributeName}]`);
+}
+
+export const TokenSelectDrawerLimit: FunctionComponent<{
+  isOpen: boolean;
+  onClose?: () => void;
+  onSelect?: (tokenDenom: string) => void;
+  showRecommendedTokens?: boolean;
+  showSearchBox?: boolean;
+  selectableAssets: (ReturnType<typeof useSwapAsset>["asset"] &
+    Partial<{
+      amount: CoinPretty;
+      usdValue: PricePretty;
+    }>)[];
+  isLoadingSelectAssets?: boolean;
+  isFetchingNextPageAssets?: boolean;
+  hasNextPageAssets?: boolean;
+  fetchNextPageAssets?: () => void;
+}> = observer(
+  ({
+    isOpen,
+    onClose: onCloseProp,
+    onSelect: onSelectProp,
+    showSearchBox = true,
+    showRecommendedTokens = true,
+    selectableAssets,
+    isLoadingSelectAssets = false,
+    isFetchingNextPageAssets = false,
+    hasNextPageAssets = false,
+    fetchNextPageAssets = () => {},
+  }) => {
+    const { t } = useTranslation();
+    const { userSettings } = useStore();
+    const { isMobile } = useWindowSize();
+    const uniqueId = useConst(() => Math.random().toString(36).substring(2, 9));
+
+    const [
+      keyboardSelectedIndex,
+      setKeyboardSelectedIndex,
+      keyboardSelectedIndexRef,
+    ] = useStateRef(0);
```

Comment:
Perhaps there's a hook we can use in `react-use` package so we don't need to define our own?

Reply:
Nothing stands out immediately but maybe a way to repurpose one of the other hooks?

This is the same as `token-select-drawer`,  Ieft this part unedited.